### ISSUE TITLE
Bug: First load/sync on new account after saving data 

### ISSUE THREAD

**Issue by zachferland** [3box/3box-js#539]
From discord, originally reported from dev branch of dapp

```
kenzo: now that were able to switch between different wallets fairly easily and quickly, the issue of profile data not fetching on first load with that account is more noticeable
```

```
michael: Basically we’re having trouble syncing new data when a box is openend for the first time after creating a new profile with some data. This is more urgent and pressing now that we’re supporting multiple providers. There will be many more first time instances. To reproduce, try to use the develop URL and create a new account with one of the providers like Portis and add data to your profile. Then refresh the page and log back in with portis and you will see that data doesn’t sync in first load, but if you refresh again it does
```

cc @oznekenzo @michaelsena

**Comment by zachferland** [comment]
Have not been able to recreated in dev dapp or else where (used in chrome). Tried with both portis and metmask. Created new accounts for both, added my name, made multiple status updates and reload page. All data was returned. Switched between accounts without issue.

**Comment by oed** [comment]
I can replicate this by doing
1. Set some data with portis
2. Clear site data
3. Reload page

Same thing seems to happen every time I clear site data.

**Comment by oed** [comment]
Seems like a bug in the dapp to me. I can replicate the same thing as above with MetaMask.

1. Set some data in example dapp
2. Clear cache in example dapp and reload
3. Login in example dapp and data loads as expected

4. In 3boxDapp login with the same MM address
5. Data doesn't load of first login
6. Reload and data appears
7. Clear cache and reload
8. Data doesn't load
9. Reload, data appears

cc @oznekenzo

**Comment by zachferland** [comment]
yeah I have seen when clearing cache though, but less frequent user path

**Comment by oznekenzo** [comment]
To add to this: 
Data loads with each provider on every account switch once that provider has had its data loaded once.  Data doesn't load when working from a blank dapp cache
